climada_petals/hazard/landslide.py

```diff
@@ -37,43 +37,88 @@
 LOGGER = logging.getLogger(__name__)
 HAZ_TYPE = 'LS'
 
-def sample_events_from_probs(prob_matrix, n_years, dist='binom'):
-    """sample an event set for a specified representative time span from
-    a matrix with annual occurrence probabilities
-    Draws events from chosen distribution.
 
+def sample_events(prob_matrix, n_years, dist='binom'):
+    """
+    Sample yearly events for a specified time span (n_years) from
+    a matrix representing annual occurrence probabilities per grid cell 
+    (prob_matrix). The matrix will usually be provided throught the
+    Landslide.from_prob() method and refer to grid-cells and their annual 
+    sliding probabilities.
+    Events are drawn from the specified distribution (binomial or 
+    poisson). 
+    Returns n_year events for the whole grid, each event being representative
+    of all slides happening in 1 year throughout the entire grid-area.
+    
     Parameters
     ----------
-    prob_matrix : scipy.sparse.csr matrix
-        matrix where each entry has an annual probability [0,1] of occurrence of
-        an event
+    prob_matrix : np.array()
+        matrix where each entry has a probability [0,1] of occurrence of
+        an event. Shape of array can be any and is determined by hazard area 
+        and resolution of probability-source-file when called through 
+        Landslide.from_prob().
     n_years : int
-        the timespan of the probabilistic simulation in years
-    dist : str
-        distribution to sample from. currently 'binom' (default) and 'poisson'
-
+        the timespan of the probabilistic simulation in years. Each year will
+        result in 1 event.
+    dist : str, 'binom' or 'poisson'
+        distribution to sample from. Default is 'binom'.
```

Comment:
Are the parameters of the binomial and Poisson distribution fixed? Then I think they should be indicated.

Reply:
no, they are given by the prob_matrix and n_samples. i added this explicitly to the description now.

Reply:
I think I am blind. Where are the parameters of binomial and Poisson characterized in the description?

Reply:
Lines #104 / 105

Reply:
In lines #104 / 105 I see no mentions of how the Expectation Value is defined.

Reply:
no idea why you need the expectation value in the random variate generator, or how this is relevant information to the user imo. don't think i need to give a crash course on statistics in a docstring, either. i rather think i don't understand what you're hinting at ^^

Reply:
Hmmm I think the misunderstanding comes from what is a sample. For instance, for the Poisson distribution and taking the signature of the corresponding scipy method:

`rvs(mu, loc=0, size=1, random_state=None)`

For me:
- size : n_samples
- mu : parameter to be specified (i.e. the expectation value)
- loc : shift of the distribution

If you set `mu = n_samples` as in the code, you will not get back `n_samples` numbers... You will get back one number which gives you the number of events given that the expectation value for events is `mu`. So, that should be reflected in the docstring imho.

Reply:
Oki. I added it accordingly.